src/client/types.ts

```diff
@@ -18,27 +23,22 @@ import type { Notification } from '../notifications/types';
 import type { ProviderToken } from '../identities/types';
 
 // Prevent overwriting the metadata type with `Omit<>`
-type ClientRPCRequestParams<T extends Record<string, JSONValue> = ObjectEmpty> =
-  {
-    metadata?: {
-      [Key: string]: JSONValue;
-    } & Partial<{
-      authorization: string;
-      timeout: number;
-    }>;
-  } & Omit<T, 'metadata'>;
+type ClientRPCRequestParams<T extends JSONObject = JSONObject> = {
+  metadata?: {
+    [Key: string]: JSONValue;
+  } & Partial<{
+    authorization: string;
+  }>;
+} & JSONRPCParams<T>;
 
 // Prevent overwriting the metadata type with `Omit<>`
-type ClientRPCResponseResult<
-  T extends Record<string, JSONValue> = ObjectEmpty,
-> = {
+type ClientRPCResponseResult<T extends JSONObject = JSONObject> = {
   metadata?: {
     [Key: string]: JSONValue;
   } & Partial<{
     authorization: string;
-    timeout: number;
   }>;
-} & Omit<T, 'metadata'>;
+} & JSONRPCResult<T>;
```

Comment:
So question about this type `JSONRPCResult`.

Shouldn't this be called `RPCResponseResult`?

It's already a forgone conclusion to be `JSON`.

Also why do we call it `RPCResponseResult`? It's because this is the result value correct?

Same reason as `RPCRequestParams` to indicate that it is the params key.

HOWVER if these are not just the `params` key` or `result` key, then we should no longer call it `XParams` or `XResult`. It can just be `RPCRequest` and `RPCResponse`.

If however there's "raw RPC" request and response, then, `JSON` prefix can be added to disambiguate.

Can you elaborate on this? @amydevs

Reply:
Ok I just read:

> Updated Type Requirements of js-rpc Request params and Response results
> The updated types in js-rpc of `JSONRPCRequestMessage`, `JSONRPCRequestNotification` and `JSONRPCResponseResult` mean that JSONRPCRequest*.params and JSONRPCResponseResult.result should now always be objects. This is so that metadata can always be appended to the params. Therefore, we need to make sure in Polykey that non-JSON-objects are never being passed as results or params.

So do you see my confusion here.

Why would there be `JSONRPCResponseResult.result`? I believe originally `JSONRPCResponseResult` IS literally the type of the `result` value inside the message as specified by the JSON RPC 2.0 spec.

If there is a `.result` property... then this type should no longer be called `JSONRPCResponseResult`. Because that doesn't make any sense!!!

Also in what situation do ever use `Message` over `Notification` or vice versa? I don't think we ever use JSON RPC 2.0 notifications right?

Reply:
i think `JSONRPCResponseResult` is currently Response with a Result. It was like that from before

Reply:
Imo, `JSONRPCResponseResult` should be renamed to `JSONRPCResponseSuccess`

Reply:
the `JSONRPCResponseResult` and `JSONRPCRequestParams` types now restrict overriding of metadata types, but still allow for user to define metadata properties as long as they don't conflict with the ones already defined (timeout)

Reply:
Where these type names changed?